bread/tests/test_public_visible.py

```diff
@@ -0,0 +1,99 @@
+import secrets
+import traceback
+
+import django.urls
+from django.contrib.admindocs.views import simplify_regex
+from django.template import TemplateDoesNotExist
+from django.test import Client, TestCase
+from django_celery_results.models import TaskResult
+from django_extensions.management.commands import show_urls
+
+from bread.contrib.reports.models import Report
+
+ROOT_URLPATTERNS = django.urls.get_resolver(None).url_patterns
+
+
+class TestAnonymousVisible(TestCase):
+    # get a list of urlpatterns
+    ALL_URL_NAMES = {
+        simplify_regex(x[1])
+        for x in show_urls.Command().extract_views_from_urlpatterns(ROOT_URLPATTERNS)
+    }
+    MODEL_OPERATIONS_REQUIRING_PK = {
+        "copy",
+        "delete",
+        "edit",
+        "read",
+    }
+
+    # (model_objects, url, additional_operations)
+    BREAD_MODELS_URLPATTERNS = (
+        (Report.objects.all(), "/reports/reports/report/", {"excel"}),
+        (TaskResult.objects.all(), "/bread/django_celery_results/taskresult/", set()),
+    )
+    EXCEPTED_URLPATTERNS = {
+        "/admin/login/",
+        "/bread/accounts/login/",
+        "/bread/accounts/password_reset/",
+        "/bread/accounts/password_reset/done/",
+        "/bread/accounts/reset/<uidb64>/<token>/",
+        "/bread/accounts/reset/done/",
+        "/bread/auth/password_reset/",
+        "/bread/auth/reset/<uidb64>/<token>/",  # password reset
+        "/bread/auth/password_reset/done/",
+        "/bread/auth/reset/done/",
+        # TODO: figure out the way to manage this view.
+        "/bread/preferences/global/<slug:section>",
```

Comment:
Here I would prefere if we use the URL name instead of a hard-coded path. This allows us to change the URL structure in the future without needing to update this test.

Reply:
For that case, both views from `accounts` and `auth` share the same names. Would it be OK if I set the namespace for Django's `auth` to `django_auth` instead, to prevent naming conflict and further difficulties?

Reply:
The django auth urls should already be inside the "admin" namespace, e.g. ```"admin:login"```. Does this conflict with bread?

Reply:
If necessary we could put the bread URLs inside a bread namespace...

Reply:
There are views that registered as the same name within `bread`. For example, `/bread/auth/password_reset/` and `/bread/accounts/password_reset/` use `password_reset`, with no any namespaces specified.

Reply:
`/bread/auth/password_reset/` is the one from `django.contrib.auth.views.PasswordResetView`, and `bread.views.auth.BreadPasswordResetView` is the one from `bread.views.auth.BreadPasswordResetView`.

Reply:
Ah, I see. Okay, we actually do not need the django-auth views anymore, we replaced everything (except the password-change, but we handle that with password-reset). You can remove ```path("auth/", include("django.contrib.auth.urls"))``` from urls.

Reply:
This one is almost completed. The only problem here is for this URL `/bread/accounts/reset/<uidb64>/<token>/`. I still have no idea how I will reverse and get the exact same path with `<` and `>` because it will be converted to the URL encoded URL.

Reply:
Did you try something like this?

```python
reverse("password_reset_confirm", kwargs={"uidb64": "hello", "toke": "world"})
```

Reply:
Okay I see. Personally I would in general try to call ```reverse``` as late as possible and work with url-names as long as possible.

For example (just pseudo-code):

```python
for urlname, urlpattern in ALL_URL_NAMES: # need to adjust creating of ALL_URL_NAMES a bit to include urlnames
    if urlname not in EXCEPTED_URLPATTERNS: # can put only urlnames inside EXCEPTED_URLPATTERNS without reverse
        urls.append(generate_url(urlname, urlpattern)) # call reverse inside generate_url
```
That way you can work with a single for-loop and you do not need to reverse the excepted URLs.

What do you think?

Reply:
Interesting. I solved this one according to the solution you had suggested using a dictionary, URL names as keys and URL patterns as values.

Reply:
I think this can be marked as resolved now.